File: roomsplit/frontend/src/types/wallet.ts
export interface Wallet {
  group_id: string
  balance_paise: number
  balance_rupees: number
  last_updated_at: string
}

export interface WalletTransaction {
  id: string
  transaction_type: 'credit' | 'debit'
  amount_paise: number
  amount_rupees: number
  balance_after_rupees: number
  description?: string
  created_at: string
}

export interface Withdrawal {
  id: string
  amount_paise: number
  amount_rupees: number
  destination_upi: string
  reason?: string
  status: 'pending' | 'approved' | 'rejected' | 'completed'
  requested_by_name: string
  approved_at?: string
  completed_at?: string
  upi_ref?: string
  created_at: string
}

File: roomsplit/backend/app/schemas/wallet.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel
from app.utils.currency import paise_to_rupees


class WalletOut(BaseModel):
    group_id: uuid.UUID
    balance_paise: int
    balance_rupees: float
    last_updated_at: datetime

    model_config = {"from_attributes": True}


class WalletTransactionOut(BaseModel):
    id: uuid.UUID
    transaction_type: str
    amount_paise: int
    amount_rupees: float
    balance_after_paise: int
    balance_after_rupees: float
    description: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}
